zerver/lib/send_email.py

```diff
@@ -96,9 +96,9 @@ def send_email(template_prefix, to_user_id=None, to_email=None, from_name=None,
 def send_email_from_dict(email_dict: Mapping[str, Any]) -> None:
     send_email(**dict(email_dict))
 
-def send_future_email(template_prefix, to_user_id=None, to_email=None, from_name=None,
+def send_future_email(template_prefix, realm, to_user_id=None, to_email=None, from_name=None,
                       from_address=None, context={}, delay=datetime.timedelta(0)):
-    # type: (str, Optional[int], Optional[Text], Optional[Text], Optional[Text], Dict[str, Any], datetime.timedelta) -> None
+    # type: (str, Realm, Optional[int], Optional[Text], Optional[Text], Optional[Text], Dict[str, Any], datetime.timedelta) -> None
```

Comment:
I think it'd be helpful to add an assert that the realm matches the to_user_id when the latter is present. Otherwise the redundancy causes a bit of worry that they could disagree, and then what?